### PR TITLE
Accept filedisk.Disk instances as well as file paths.

### PR DESCRIPTION
**Pull request by zvin**
https://www.npmjs.com/package/file-disk

### CODE REVIEW

File: lib/boot-record.coffee
Comment (zvin) [inline lib/boot-record.coffee]: @jviotti how do I specify that the first parameter can be either a String or a filedisk.Disk ?

**Comment by lurch** [comment]
@jhermsmeier Does this overlap with some of the stuff that you've been doing recently??

File: lib/boot-record.coffee
Comment (jhermsmeier) [reply lib/boot-record.coffee]: @zvin `@param {String|filedisk.Disk} image` should do the trick

**Comment by jhermsmeier** [comment]
@lurch I guess this could be done with [disk](https://github.com/jhermsmeier/node-disk) and/or [mbr](https://github.com/jhermsmeier/node-mbr), [gpt](https://github.com/jhermsmeier/node-gpt) etc. as well – but that ship has already sailed, from what I can see (meaning there'd need to be quite some API-bending). Not something to change in this PR.

**Comment by lurch** [comment]
Sorry, I didn't mean this PR specifically (sorry I didn't make that clear) I meant this repo in general...

**Comment by jhermsmeier** [comment]
And I guess `file-disk` is what I use my [blockdevice](https://github.com/jhermsmeier/node-blockdevice) for – which is the base API which my disk image formats implement – so that you can use [disk](https://github.com/jhermsmeier/node-disk) or other modules with any other module implementing a format or actual block device without having to change anything.

**Review by jviotti — APPROVED** [review]
Approving as long as the tests pass. Feel free to merge and publish yourself :)

**Comment by lurch** [comment]
CI builds are failing with e.g.
```
2.46s$ nvm install 0.12
Downloading and installing node v0.12.18...
Downloading https://nodejs.org/dist/v0.12.18/node-v0.12.18-linux-x64.tar.xz...
######################################################################## 100.0%
Computing checksum with sha256sum
Checksums matched!
Now using node v0.12.18 (npm v2.15.11)
$ node --version
v0.12.18
$ npm --version
2.15.11
$ nvm --version
0.33.2
before_install
13.56s$ npm install -g npm@latest
/home/travis/.nvm/versions/node/v0.12.18/bin/npm -> /home/travis/.nvm/versions/node/v0.12.18/lib/node_modules/npm/bin/npm-cli.js
npm@5.0.1 /home/travis/.nvm/versions/node/v0.12.18/lib/node_modules/npm
4.24s$ npm install 
npm WARN npm npm does not support Node.js v0.12.18
```

I guess the tests themselves need some TLC before this PR can be merged?

**Pull request merged**
